Assets/MixedRealityToolkit/_Core/Devices/OpenVR/GenericOpenVRController.cs

```diff
@@ -138,211 +138,237 @@ private void UpdateControllerData(XRNodeState State, MixedRealityInteractionMapp
-            //                }
-            //                else
-            //                {
-            //                    InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
-            //                }
-            //            }
-            //            break;
-            //        }
-            //    case DeviceInputType.Trigger:
-            //        {
-            //            //Update the interaction data source
-            //            interactionMapping.SetFloatValue(interactionSourceState.selectPressedAmount);
-
-            //            // If our value changed raise it.
-            //            if (interactionMapping.Changed)
-            //            {
-            //                //Raise input system Event if it enabled
-            //                InputSystem?.RaiseOnInputPressed(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionSourceState.selectPressedAmount);
-            //            }
-            //            break;
-            //        }
-            //    default:
-            //        throw new IndexOutOfRangeException();
-            //}
+            switch (interactionMapping.InputType)
+            {
+                case DeviceInputType.TriggerPress:
+                case DeviceInputType.Select:
+                    {
+                        //Get the current Trigger button state
+                        var triggerButton = ControllerHandedness == Handedness.Left ? Input.GetKey(KeyCode.JoystickButton14) : Input.GetKey(KeyCode.JoystickButton15);
+
+                        //Update the interaction data source
+                        interactionMapping.SetBoolValue(triggerButton);
+
+                        // If our value changed raise it.
+                        if (interactionMapping.Changed)
+                        {
+                            Debug.LogWarning($"Trigger pressed for [{ControllerHandedness}] hand, set to [{triggerButton}]");
+
+                            //Raise input system Event if it enabled
+                            if (triggerButton)
+                            {
+                                InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                            }
+                            else
+                            {
+                                InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                            }
+                        }
+                        break;
+                    }
+                case DeviceInputType.Trigger:
+                    {
+                        //Get the current Trigger axis state - ** Does not WORK
+                        var triggerAxis = ControllerHandedness == Handedness.Left ? Input.GetAxis("axis 9") : Input.GetAxis("axis 10");
```

Comment:
Not sure about having to do this by Axis. We can get buttons fine, but axes are still something users have to setup by hand, which can be annoying.

Reply:
This was a rage check-in as the above does not work.  As per the updated descripting, I'm really hating Unity input right now.